x-pack/solutions/search/plugins/enterprise_search/server/deprecations/routes.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { deleteConnectorById, putUpdateNative } from '@kbn/search-connectors';
+
+import { RouteDependencies } from '../plugin';
+
+import { elasticsearchErrorHandler } from '../utils/elasticsearch_error_handler';
+
+export function registerDeprecationRoutes({ router, log }: RouteDependencies) {
+  router.post(
+    {
+      path: '/internal/enterprise_search/deprecations/delete_crawler_connectors',
+      validate: {
+        body: schema.object({
+          ids: schema.arrayOf(schema.string()),
+          deprecationDetails: schema.object({ domainId: schema.literal('enterpriseSearch') }),
+        }),
+      },
+    },
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const { client } = (await context.core).elasticsearch;
+      for (const connectorId of request.body.ids) {
+        await deleteConnectorById(client.asCurrentUser, connectorId);
+      }
+      return response.ok({ body: { deleted: request.body.ids } });
+    })
+  );
+
+  router.post(
+    {
+      path: '/internal/enterprise_search/deprecations/convert_connectors_to_client',
+      validate: {
+        body: schema.object({
+          ids: schema.arrayOf(schema.string()),
+          deprecationDetails: schema.object({ domainId: schema.literal('enterpriseSearch') }),
+        }),
+      },
+    },
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const { client } = (await context.core).elasticsearch;
+      for (const connectorId of request.body.ids) {
+        await putUpdateNative(client.asCurrentUser, connectorId, false);
+      }
```

Comment:
nit: how many of these ids do we expect? any chances this times out on a big list? as written this will run the `putUpdateNative` on after another which would not be performant for any large lists.

This code would run all the calls together instead of on after another. but if any of them throws an error it will get raised up to the error handler.

```suggestion
      await Promise.all(
          request.body.ids.map((connectorId) =>  putUpdateNative(client.asCurrentUser, connectorId, false)),
      );
```

Reply:
Not many. Though, if it did time out, they could click the button to try again, and it wouldn't re-do the ones that it successfully processed before.

This is engineering for an edge case. No one _should_ have connectors in a state that matches this deprecation, we just know from telemetry that people _have_ created native connectors with "foobar" service types. 

But I'm happy to make the suggested edit, looks straightforward.